classes/cyclonedx-export.bbclass

```diff
@@ -77,48 +100,26 @@ python do_cyclonedx_package_collect() {
-                    })
-                else:
-                    vex["vulnerabilities"][index_found]["affects"].append(
-                        {"ref": f"urn:cdx:{sbom_serial_number}/1#{bom_ref}"}
-                    )
-
-            # populate vex file with ignored CVEs defined in CVE_CHECK_IGNORE
-            cve_check_ignore = d.getVar("CVE_CHECK_IGNORE")
-            if cve_check_ignore is not None:
-                for ignored_cve in cve_check_ignore.split():
-                    bb.debug(2, f"Found ignore statement for CVE {ignored_cve} in {name}@{version}")
-                    index_found = next((i for i, v in enumerate(vex["vulnerabilities"]) if v["id"] == ignored_cve), None)
-                    if index_found is None:
-                        vex["vulnerabilities"].append({
-                            "id": ignored_cve,
-                            # vex documents require a valid source, see https://github.com/DependencyTrack/dependency-track/issues/2977
-                            # this should always be NVD for yocto CVEs.
-                            "source": {"name": "NVD", "url": f"https://nvd.nist.gov/vuln/detail/{ignored_cve}"},
-                            # setting not-affected state for ignored CVEs
-                            "analysis": {"state": "not_affected"},
-                            # Hint: Component specific resolving seems not to work at the moment when using DependencyTrack
-                            # resolution will of CVE will be applied to all components within the project that contain the CVE
-                            "affects": [{"ref": f"urn:cdx:{sbom_serial_number}/1#{bom_ref}"}]
-                        })
-                    else:
-                        vex["vulnerabilities"][index_found]["affects"].append(
-                            {"ref": f"urn:cdx:{sbom_serial_number}/1#{bom_ref}"}
-                        )
+            # populate vex file with details from CVE_STATUS
+            cve_ignored = []
+            cve_patched = []
+            for cve in (d.getVarFlags("CVE_STATUS") or {}):
+                decoded_status, _, _ = decode_cve_status(d, cve)
+                if decoded_status == "Patched":
+                    cve_patched.append(cve)
+                if decoded_status == "Ignored":
+                    cve_ignored.append(cve)
+
+            for cve in cve_patched:
+                bb.debug(2, f"Found patch for CVE {cve} in {name}@{version}")
+                # map Yoctos "Patched" to CycloneDXs "resolved"
+                state = "resolved"
+                append_to_vex_vulnerabilities(d, vex, cve, state, sbom_serial_number, bom_ref)
+            for cve in cve_ignored:
+                bb.debug(2, f"Found ignore statement for CVE {cve} in {name}@{version}")
+                # map Yoctos "Ignored" to CycloneDXs "not_affected"
+                state = "not_affected"
+                append_to_vex_vulnerabilities(d, vex, cve, state, sbom_serial_number, bom_ref)
```

Comment:
```
for cve in (d.getVarFlags("CVE_STATUS") or {}):
  append_to_vex_vulnerabilities(d, vex, cve, sbom_serial_number, bom_ref)
```

and later :
```
def append_to_vex_vulnerabilities(d, vex, cve, sbom_serial_number, bom_ref):
  state, detail, description = decode_cve_status(d, cve)
  if state == "Patched":
    bb.debug(...)
    state = "resolved"
  elif state == "Ignored":
    bb.debug(...)
    state = "not_affected"
  else:
    return

  ...
```

Reply:
fair :sweat_smile:

Reply:
Pushed an update, haven't tested yet though.